vspec2proto.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+
+#
+# (c) 2016 Jaguar Land Rover
+# (c) 2021 Robert Bosch GmbH
+#
+# All files and artifacts in this repository are licensed under the
+# provisions of the license provided by the LICENSE file in this repository.
+#
+#
+# Concert vspec file to JSON
+#
+
+import sys
+import vspec
+import json
+import getopt
+import re
+
+from model.vsstree import VSSNode, VSSType
+
+
+def usage():
+    print("Usage:", sys.argv[0], "[-I include_dir] ... [-i prefix:id_file] vspec_file proto_file")
+    print("  -I include_dir       Add include directory to search for included vspec")
+    print("                       files. Can be used multiple timees.")
+    print()
+    print("  -i prefix:uuid_file  File to use for storing generated UUIDs for signals with")
+    print("                       a given path prefix. Can be used multiple times to store")
+    print("                       UUIDs for signal sub-trees in different files.")
+    print()
+    print(" vspec_file            The vehicle specification file to parse.")
+    print(" json_file             The file to output the JSON objects to.")
+    sys.exit(255)
+
+
+
+def getDataType(node):
+    um =  re.compile('uint.*')
+    sm =  re.compile('int.*')
+    m64 = re.compile('.int64')
+
+    if str(node.data_type.value) == "boolean":
+        return "bool"
+    
+    elif m64.match(str(node.data_type.value)) != None:
+        return str(node.data_type.value)
+        
+    elif um.match(str(node.data_type.value)) != None:
+        return "uint32"
+
+    elif sm.match(str(node.data_type.value)) != None:
+        return "int32"
+           
+    elif str(node.data_type.value) == "float":
+        return "float"
+    elif str(node.data_type.value) == "double":
+        return "double"
+    elif str(node.data_type.value) == "string":
+        return "string"
+    elif str(node.data_type.value) == "string[]": #probably need more generic array handling
+        return "repeated string"
+    else:
+        print("Unsupported datatype "+str(node.data_type.value))
+        return None
+
+
+
+def createMessage(file, prefix, node):
```

Comment:
This method body can start with:
`file.write("message "+prefix+" {\n")`
and end with:
`file.write("}\n\n")`

This will be written to the file either way. Currently both of these write statements exist in the if clause (in createMessage) and in the else clause of this method. By putting it here, outside the if clause, it will only be needed once.